### PR TITLE
Start new checkpoint if needed after current checkpoint has completed

### PR DESCRIPTION
**Pull request by saiseekerhn**
I changed the where the counts were reset so that Checkpointer.scala resets counts when starting a checkpoint rather than after

### CODE REVIEW

**Comment by Topher-the-Geek** [comment]
Add to whitelist.

**Comment by Topher-the-Geek** [comment]
Cool. It passed the tests. It's not quite there yet though. When one checkpoint completes, the Checkpointer needs to check if it should start the next. It should disengage only if it doesn't need to start the next checkpoint.

There's wasn't test for the Checkpointer, and writing one involves lots of fiddling to bring a number of pieces together. So I've [started a CheckpointSpec](https://github.com/Topher-the-Geek/store/commit/576d0df80813b302d9109781fc9f4d2a50422cf6) to get you going. There's two tests there, and the second one fails. It will pass once you make the additional change that's needed.

You should pull my new test into your repository, and then take it from there. Also, be sure to pull recent changes from the master branch in Treode/store into your repository.

**Comment by saiseekerhn** [comment]
So I pulled recent changes from Treode/store. But it seems like CheckpointSpec.scala is in your branch.
I am not too familiar with github, but should I set up remote upstream to your repo as well to pull your changes?

**Comment by Topher-the-Geek** [comment]
Indeed, that's the way to do it.

**Comment by saiseekerhn** [comment]
Should I be merging my master only with the upstream/checkpoint-test since that is where the test lives? Also will this erase my previous bug fix, or does it only merge your updates into mine?

**Comment by saiseekerhn** [comment]
got it

**Comment by Topher-the-Geek** [comment]
Yep. You got the all the changes merged. Awesome. Now, can you fix the test? Drop into the chat room if you have questions.

**Comment by saiseekerhn** [comment]
I committed my changes to my branch. I don't need to create a pull request for that again right?

**Comment by saiseekerhn** [comment]
just realized I didn't set checkreqs = List.empty before the if statement in my latest commit. so if another checkpoint needs to be done checkreqs won't be empty I think. I am not entirely sure I understand its role, but I am guessing since the first line of reengage uses checkreqs, I should probably set that to empty and then perform the checkpoint?

**Comment by Topher-the-Geek** [comment]
That's a different can of worms. We will address it in later CLs. As long
as the entire test suite passes for this change, it's good.
--

File: disk/src/com/treode/disk/Checkpointer.scala [hunk 1]
```diff
@@ -36,15 +36,18 @@ private class Checkpointer (kit: DiskKit) {
...
     fanout (checkreqs) .pass (())
-    checkreqs = List.empty
-    bytes = 0
-    entries = 0
-    engaged = false
+    if (config.checkpoint (this.bytes, this.entries))
+      _checkpoint()
+    else
+      checkreqs = List.empty
+      engaged = false
```
Comment (Topher-the-Geek) [inline disk/src/com/treode/disk/Checkpointer.scala]: I got a chance to look closely at this CL this morning.  Given your exposure to Python, your assumption here is reasonable, but it's wrong. :-)

    if (...)
       then-clause
    else
       else-clause
       oops

Python (and Haskell, and others) would use the indentation to infer that In `oops` is part of the else clause. However Scala does not, and the above code is like this:

    if (...) {
        then-clause
    } else {
        else-clause
    }
    oops

Could you fix that. Set `engaged` to false in the else clause. Leave the assignment to `checkreqs` where it originally was.

**Comment by saiseekerhn** [comment]
Oops, I just realized you can leave out the braces if the if statement is followed by only one line. Fixed it now. Thanks for catching that.

**Pull request closed without merging**